zeebe/engine/src/test/java/io/camunda/zeebe/engine/util/client/ClusterVariableClient.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.util.client;
+
+import io.camunda.zeebe.protocol.impl.record.value.variable.ClusterVariableRecord;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.intent.ClusterVariableIntent;
+import io.camunda.zeebe.protocol.record.value.ClusterVariableRecordValue;
+import io.camunda.zeebe.test.util.MsgPackUtil;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import java.util.Random;
+import java.util.function.LongFunction;
+import org.agrona.concurrent.UnsafeBuffer;
+
+public final class ClusterVariableClient {
+
+  private static final long DEFAULT_KEY = -1;
+
+  private static final LongFunction<Record<ClusterVariableRecordValue>> SUCCESS_SUPPLIER =
+      (sourceRecordPosition) ->
+          RecordingExporter.clusterVariableRecords()
+              .onlyEvents()
+              .withSourceRecordPosition(sourceRecordPosition)
+              .getFirst();
+  private static final LongFunction<Record<ClusterVariableRecordValue>> REJECTION_SUPPLIER =
+      (sourceRecordPosition) ->
+          RecordingExporter.clusterVariableRecords()
+              .onlyCommandRejections()
+              .withSourceRecordPosition(sourceRecordPosition)
+              .getFirst();
+
+  private final long requestId = new Random().nextLong();
+  private final int requestStreamId = new Random().nextInt();
+
+  private final ClusterVariableRecord clusterVariableRecord;
+  private final CommandWriter writer;
+  private LongFunction<Record<ClusterVariableRecordValue>> expectation = SUCCESS_SUPPLIER;
+
+  public ClusterVariableClient(final CommandWriter writer) {
+    clusterVariableRecord = new ClusterVariableRecord();
+    this.writer = writer;
+  }
+
+  public ClusterVariableClient withValue(final Object value) {
+    clusterVariableRecord.setValue(new UnsafeBuffer(MsgPackUtil.asMsgPack(value)));
+    return this;
+  }
```

Comment:
```suggestion
  public ClusterVariableClient withValue(final String value) {
    return withValue(new UnsafeBuffer(MsgPackConverter.convertToMsgPack(value)));
  }

  public ClusterVariableClient withValue(final DirectBuffer value) {
    clusterVariableRecord.setValue(value);
    return this;
  }
```

Aligning this with the `VariableClient` to avoid implementing additional utilities.